django/contrib/auth/mixins.py

```diff
@@ -57,6 +56,8 @@ def handle_no_permission(self):
             not login_netloc or login_netloc == current_netloc
         ):
             path = self.request.get_full_path()
+        from django.contrib.auth.views import redirect_to_login
```

Comment:
We already do a local import to avoid circular import [here](https://github.com/django/django/blob/549320946dfdaac6ccb6c0d3150c46db92637187/django/contrib/auth/decorators.py#L34). We can also move `redirect_to_login` to `utils.py` and follow a [similar](https://github.com/django/django/commit/98e05ccde440cc9b768952cc10bc8285f4924e1f#diff-76917634c6c088f56f8dec9493294c657953e61b01e38e33b02d876d5e96dd3aR7-R8) approach for backwards compatibility. What do you prefer?

Reply:
Another instance of avoiding a circular import sounds like more reason to move `redirect_to_login` but this change can come in a separate PR.

Reply:
Moved it [here](https://github.com/django/django/pull/18072)

Reply:
Commented on the PR, but I checked with other contributors is they thought this was the right approach. As this is a change to a documented stable API, decided the benefit is not strong enough to justify the depreciation 
But thank you again for your efforts here, very appreciated